EU4toV2/Source/V2World/Country/Country.h

```diff
@@ -0,0 +1,166 @@
+		int getCapital() const { return details.capital; }
+		void addState(std::shared_ptr<State> newState, const mappers::PortProvinces& portProvincesMapper);
+		void convertLeaders(mappers::LeaderTraitMapper& leaderTraitMapper);
+		void convertUncivReforms(int techGroupAlgorithm, double topTech, int topInstitutions, const mappers::TechGroupsMapper& techGroupsMapper);
+		void addRailroadtoCapitalState();
+		void addTech(const std::string& newTech) { techs.insert(newTech); }
+		void convertLandlessReforms(std::shared_ptr<Country> capOwner);
+		const UncivReforms& getUncivReforms() const { return uncivReforms; }
+		void setArmyTech(double normalizedScore);
+		void setNavyTech(double normalizedScore);
+		void setCommerceTech(double normalizedScore);
+		void setIndustryTech(double normalizedScore);
+		void setCultureTech(double normalizedScore);
+		bool addFactory(std::shared_ptr<Factory> factory);
+		std::string getCommonCountryFile() const { return commonCountryFile; }
+		std::string getLocalName() const { return localisation.getLocalName(); };
+		std::string getReligion() const { return details.religion; }
+		std::string getColonialRegion();
+		const Localisation& getLocalisation() const { return localisation; }
+		bool isDynamicCountry() const { return dynamicCountry; }
+		std::string getFilename() const { return details.filename; }
+		static bool factoryCandidateSortPredicate(const std::pair<double, std::shared_ptr<State>>& lhs, const std::pair<double, std::shared_ptr<State>>& rhs);
+
+		friend std::ostream& operator<<(std::ostream& output, const Country& country);
+		void outputCommons(std::ostream& output);
+		void outputOOB(std::ostream& output);
+		
+	private:
+		bool dynamicCountry = false;	// true if this country is a Vic2 dynamic country
+		bool newCountry = false; // true if this country is being added by the converter, i.e. doesn't already exist in Vic2
+		mappers::CountryDetails details;
+		std::string tag;
+		std::string commonCountryFile;
+		Localisation localisation;
+		std::shared_ptr<EU4::Country> srcCountry;
+		UncivReforms uncivReforms;
+		Reforms reforms;
+		std::shared_ptr<Country> colonyOverlord;
+		std::map<std::string, Relation> relations;
+		std::map<int, std::shared_ptr<Province>> provinces;
+		std::set<std::string> decisions;
+		EU4::NationalSymbol nationalColors;
+		std::vector<std::shared_ptr<State>> states;
+		std::vector<Leader> leaders;
+		std::set<std::string> techs;
+		std::set<std::string> inventions;
+		std::map<REGIMENTTYPE, int> unitNameCount;
+		std::map<REGIMENTTYPE, double> countryRemainders;
+		std::vector<Army> armies;
+
+		void loadPartiesFromBlob(const mappers::PartyNameMapper& partyNameMapper, const mappers::PartyTypeMapper& partyTypeMapper);
+		void initParties(const mappers::PartyNameMapper& partyNameMapper, const mappers::PartyTypeMapper& partyTypeMapper);
+		void setReligion(std::shared_ptr<EU4::Country> srcCountry, const mappers::ReligionMapper& religionMapper);
```

Comment:
Can pass a const reference to the source country instead of the pointer (both shorter signature and better const correctness for the source country).

Applies to other similar items.

Reply:
passing pointer to country instead of religion string. fixed.